schemas/xml/README.md

```diff
@@ -1,57 +1,449 @@
+
+Let us make an example.
+The example will be agnostic of the particular meta-model version, so that it can age well across different versions.
+We fantasize a meta-model class `SomeAbstractClass` and assume it has two descendant classes, `SomeConcreteClass` and `AnotherConcreteClass`.
+Let us assume yet another class, `YetAnotherClass`, entails the attribute `someAttribute` of type `SomeAbstractClass`.
+
+Here is how the XML structure corresponding to `YetAnotherClass/someAttribute` would look like in this fantasized case, where the value is an instance of `SomeConcreteClass`:
+
+```xml
+
+<someAttribute>
+    <SomeConcreteClass>
+        ...
+        serialized
+        attributes
+        of
+        SomeConcreteClass
+        ...
+    </SomeConcreteClass>
+</someAttribute>
+```
+
+If the value is an instance of `AnotherConcreteClass`, the serialization becomes:
+
+```xml
+
+<someAttribute>
+    <AnotherConcreteClass>
+        <!-- ... serialized attributes of AnotherConcreteClass ... -->
+    </AnotherConcreteClass>
+</someAttribute>
+```
+
+The abstract class, `SomeAbstractClass`, does not show up in the serialization at all, as it is redundant information.
+
+#### Instances Serialized without Discriminator
+
+If the concrete type of the attribute at deserialization is unambiguous by the meta-model, we omit the discriminator to reduce the clutter.
+The instance is simply serialized as a sequence of XML elements corresponding to its attributes.
+
+Let us fantasize yet another example, similar to the one in Section [Instances Serialized with Discriminator].
+We will again draw an example such that it is agnostic of meta-model version for better evolution of this Readme.
+Assume a class `SomeClass` with an attribute `SomeClass/someAttribute`.
+Now imagine the type of `someAttribute` to be the class `AnotherClass`.
+The class `AnotherClass` provides attributes `AnotherClass/anotherAttribute` and `AnotherClass/yetAnotherAttribute`.
+The class `AnotherClass` has no descendants, so the concrete type of `SomeClass/someAttribute` is unambiguous.
+
+[Instances Serialized with Discriminator]: #Instances-Serialized-with-Discriminator
+
+Here is how the XML structure would look like:
+
+```xml
+
+<someAttribute>
+    <anotherAttribute>
+        <!-- ... -->
+    </anotherAttribute>
+    <yetAnotherAttribute>
+        <!-- ... -->
+    </yetAnotherAttribute>
+</someAttribute>
+```
+
+The type information about `AnotherClass` is omitted, as the type of the `SomeClass/someAttribute` is fixed in the meta-model.
+
+### Attribute Values as Aggregations
+
+Many attributes in the meta-model do not represent a single value (be it primitive or structured as a class), but aggregate instances of meta-model classes.
+For example, [Submodel/submodelElement] aggregates instances of [SubmodelElement]'s.
+
+If we just concatenated all the properties of the instances, we would not which property belongs to which instance (or that would be complicated).
```

Comment:
editorial: we would not know - know is missing

Reply:
Fixed.